File: dotnet/src/Manager.Api/Authorization/AdminResourceAccess.cs
namespace Manager.Api.Authorization;

/// <summary>
/// Manager API treats some features as admin resource management:
/// JWT is required, but handlers do not scope rows to <c>IUserContext.UserId</c>.
/// </summary>
public static class AdminResourceAccess
{
    private static readonly HashSet<string> AdminManagedFeatures =
        new(StringComparer.Ordinal) { "Users" };

    public static bool IsAdminManagedFeature(string featureName) =>
        AdminManagedFeatures.Contains(featureName);

    public static bool RequiresResourceOwnership(string featureName) =>
        !IsAdminManagedFeature(featureName);
}

File: python/src/authorization/admin_resource_access.py
from __future__ import annotations

_ADMIN_MANAGED = frozenset({"Users"})


def is_admin_managed_feature(feature_name: str) -> bool:
    return feature_name in _ADMIN_MANAGED


def requires_resource_ownership(feature_name: str) -> bool:
    return feature_name not in _ADMIN_MANAGED
